test/bats/test.bats

```diff
@@ -475,6 +475,30 @@ EOF
   assert_success
   wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl wait --for=condition=Ready --timeout=60s pod -l run=dummy-provider -n ${GATEKEEPER_NAMESPACE}"
 
+  # status test - wait for providerpodstatus to be created and verify content
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl get providerpodstatus -n ${GATEKEEPER_NAMESPACE} --no-headers | wc -l | grep -E '^[1-9][0-9]*$'"
+  
+  # verify at least one providerpodstatus exists with the correct provider label
+  run kubectl get providerpodstatus -n ${GATEKEEPER_NAMESPACE} -l internal.gatekeeper.sh/provider-name=dummy-provider --no-headers
+  assert_success
+  [[ $(echo "${output}" | wc -l) -ge 1 ]]
+  
+  # verify the providerpodstatus has correct status fields
+  run kubectl get providerpodstatus -n ${GATEKEEPER_NAMESPACE} -l internal.gatekeeper.sh/provider-name=dummy-provider -o jsonpath='{.items[0].status.active}'
+  assert_success
+  assert_match "true" "${output}"
+  
+  # verify the provider itself shows status information
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl get provider dummy-provider -o jsonpath='{.status.byPod}' | grep -v '\\[\\]'"
+
+  # metrics test
+  kubectl run temp --image=curlimages/curl -- tail -f /dev/null
+  kubectl wait --for=condition=Ready --timeout=60s pod temp
+
+  local pod_ip="$(kubectl -n ${GATEKEEPER_NAMESPACE} get pod -l gatekeeper.sh/operation=webhook -ojson | jq --raw-output '[.items[].status.podIP][0]' | sed 's#\.#-#g')"
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl exec -it temp -- curl http://${pod_ip}.${GATEKEEPER_NAMESPACE}.pod:8888/metrics | grep 'gatekeeper_provider'"
```

Comment:
do we want to grep both metrics?

Reply:
`gatekeeper_provider_error_count_total` metric is not reported until there is an actual error, and its hard to replicate `conversion` and `upsert` errors. I tested this in local with manually incresing the count and metrics is reported as expected.